api/src/main/java/io/druid/data/input/impl/FileIteratingFirehose.java

```diff
@@ -67,6 +67,7 @@ public InputRow nextRow()
         }
 
         lineIterator = lineIterators.next();
+        parser.reset();
```

Comment:
Please reset the parser at [here](https://github.com/druid-io/druid/pull/4198/files#diff-96eabf509a270f600bffcd629f42906bR53) too. Also, lineIterator must be closed before assigning new one there.